privacy/anonymizing-your-mac-address.md

```diff
@@ -170,3 +172,48 @@ To disable MAC Randomizing if you find that a network connecting to does not lik
 - Also in `fedora-23` type `sudo systemctl disable macspoof@enp0s0`
 - Remove the service file `sudo rm /etc/systemd/system/macspoof@.service` in TemplateVM
 - Delete the package `sudo dnf remove macchanger`
+
+Method 2
+========
+
+An alternative approach is to use udev rules.
+
+This method does not require you to specify interface names, will apply to all interfaces, even those added after boot, and will retain the random MAC address through sleep/wake cycles.
+
+Install **macchanger** package in the TemplateVM backing sys-net.
+
+In that template create a udev rule using an editor of your choice.
+
+In /etc/udev/rules.d create the file 00-mac-spoof.rules with content:
+```
+SUBSYSTEM=="net", ACTION=="add", ATTR{type}=="1", RUN+="/usr/bin/macchanger -e  $name"
```

Comment:
> @marmarek Is there a way to tell internally if a device is real hardware?

I've tried to find some way but found only some workarounds. Generally take a look what udev shows about those devices:
```
$ udevadm info -q all /sys/class/net/eth0
P: /devices/vif-0/net/eth0
E: DEVPATH=/devices/vif-0/net/eth0
E: ID_MM_CANDIDATE=1
E: ID_NET_NAME_MAC=enx00163e5e6c17
E: ID_OUI_FROM_DATABASE=Xensource, Inc.
E: ID_PATH=xen-vif-0
E: ID_PATH_TAG=xen-vif-0
E: IFINDEX=2
E: INTERFACE=eth0
E: SUBSYSTEM=net
E: SYSTEMD_ALIAS=/sys/subsystem/net/devices/eth0
E: TAGS=:systemd:
E: USEC_INITIALIZED=2699433
```
```
$ udevadm info -q all /sys/class/net/vif2.0 
P: /devices/vif-2-0/net/vif2.0
E: DEVPATH=/devices/vif-2-0/net/vif2.0
E: ID_MM_CANDIDATE=1
E: ID_NET_DRIVER=vif
E: ID_NET_LINK_FILE=/usr/lib/systemd/network/99-default.link
E: ID_NET_NAME_MAC=enxfeffffffffff
E: IFINDEX=5
E: INTERFACE=vif2.0
E: SUBSYSTEM=net
E: SYSTEMD_ALIAS=/sys/subsystem/net/devices/vif2.0
E: TAGS=:systemd:
E: USEC_INITIALIZED=441230323
```

Maybe `ID_PATH!="xen-vif-*", ID_NET_DRIVER!="vif"` would be enough?